src/basic/escape.c

```diff
@@ -119,7 +119,7 @@ char *cescape(const char *s) {
         return cescape_length(s, strlen(s));
 }
 
-int cunescape_one(const char *p, size_t length, char *ret, uint32_t *ret_unicode) {
+int cunescape_one(const char *p, size_t length, char *ret, char32_t *ret_unicode) {
```

Comment:
I see this only now, but `ret_unicode` is never actually accessed anywhere in this function.
@poettering ?